tool/tsh/common/vnet_windows.go

```diff
@@ -17,95 +17,82 @@
 package common
 
 import (
-	"fmt"
-	"os"
-
 	"github.com/alecthomas/kingpin/v2"
 	"github.com/gravitational/trace"
+	"golang.org/x/sys/windows/svc"
 
-	"github.com/gravitational/teleport"
-	"github.com/gravitational/teleport/api/types"
 	"github.com/gravitational/teleport/lib/vnet"
-	"github.com/gravitational/teleport/lib/vnet/daemon"
 )
 
-type vnetCommand struct {
-	*kingpin.CmdClause
+func isWindowsService() bool {
+	isSvc, err := svc.IsWindowsService()
+	return err == nil && isSvc
 }
 
-func newVnetCommand(app *kingpin.Application) *vnetCommand {
-	cmd := &vnetCommand{
-		CmdClause: app.Command("vnet", "Start Teleport VNet, a virtual network for TCP application access.").Hidden(),
+func newVnetCommands(app *kingpin.Application) *vnetCommands {
+	if isWindowsService() {
+		return &vnetCommands{
+			subcommands: []vnetCLICommand{
+				newVnetServiceCommand(app),
+			},
+		}
+	}
+	return &vnetCommands{
+		subcommands: []vnetCLICommand{
+			newVnetCommand(app),
+			newVnetInstallServiceCommand(app),
+			newVnetUninstallServiceCommand(app),
```

Comment:
Connect will have to perform its own equivalent of those commands, right?

Reply:
no, the vnet service command will run when tshd re-execs itself, it should be handled by tsh fine